test/unit/foreman/templates/snippets/kickstart_network_interface_test.rb

```diff
@@ -0,0 +1,325 @@
+      assert_match(/--netmask/, actual)
+      assert_match(/--gateway/, actual)
+      assert_match(/--bootproto/, actual)
+      assert_match(/static/, actual)
+    end
+
+    test 'should use dhcp ipv4 configuration' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        subnet: FactoryBot.build(:subnet_ipv4_with_domains)
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/--bootproto/, actual)
+      assert_match(/dhcp/, actual)
+    end
+
+    test 'should use static ipv6 configuration' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        subnet6: FactoryBot.build(:subnet_ipv6_static_for_snapshots)
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: true
+      )
+
+      assert_match(/--ipv6=/, actual)
+      assert_match(/--ipv6gateway=/, actual)
+    end
+
+    test 'should use dhcp ipv6 configuration' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        subnet6: FactoryBot.build(:subnet_ipv6_with_domains)
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/--ipv6/, actual)
+      assert_match(/dhcp/, actual)
+    end
+
+    test 'should use auto ipv6 configuration' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        subnet6: FactoryBot.build(:subnet_ipv6_dhcp_for_snapshots)
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: true,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/--ipv6/, actual)
+      assert_match(/auto/, actual)
+    end
+
+    test 'should set vlan options' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        virtual: true,
+        tag: '333',
+        attached_to: 'test_iface1'
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/--vlanid/, actual)
+      assert_match(/333/, actual)
+      assert_match(/--interfacename/, actual)
+    end
+
+    test 'should set DNS servers' do
+      iface = FactoryBot.build(
+        :nic_managed,
+        primary: true,
+        subnet: FactoryBot.build(:subnet_ipv4_static_for_snapshots)
+      )
+
+      actual = render_template(
+        iface,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/--nameserver/, actual)
+      assert_match(/192.168.42.2/, actual)
+      assert_match(/192.168.42.3/, actual)
```

Comment:
Ideally a test only has a single assertion. This is a much stricter check. It's showing up more often in this file, so only commenting once.
```suggestion
      assert_includes(actual, '--nameserver=192.168.42.2,192.168.42.3')
```